soc: nrf52: validate_enabled_instances fixes

mbolivar-nordic opened the pull request:
UART0, TWI0, and SPI0 have different peripheral IDs (2, 3, and 4
respectively) on this SoC and therefore should not be mutually
exclusive to use.

mbolivar-nordic committed:
soc: nrf52805: fix enabled instance 0 check

UART0, TWI0, and SPI0 have different peripheral IDs (2, 3, and 4
respectively) on this SoC and therefore should not be mutually
exclusive to use.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic commented:
@ioannisg @carlescufi CI failures are unrelated; please merge

ioannisg reviewed (commented):
how about 52811, @mbolivar-nordic ?

mbolivar-nordic commented:
> how about 52811, @mbolivar-nordic ?

I got a report from an FAE only about this SoC. I'm not trying to audit all of the SoCs and fix the file for them, just trying to address this particular bug report.

Are you saying you've checked everything and 52811 is the only other one missing?

ioannisg commented:
> > how about 52811, @mbolivar-nordic ?
> 
> I got a report from an FAE only about this SoC. I'm not trying to audit all of the SoCs and fix the file for them, just trying to address this particular bug report.
> 
> Are you saying you've checked everything and 52811 is the only other one missing?

Did't check, but, afaik the three "low-end" nrf52x variants we have are 52805, 52810, 52811, so it is worth checking one more SoC.

carlescufi commented:
@anangl could you please check that?

mbolivar-nordic committed:
soc: nrf52811: fix enabled instance 0 check

Similar fix to that for the 52805.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic commented:
@ioannisg I added a fix for 52811 also.

anangl commented on soc/arm/nordic_nrf/validate_enabled_instances.c:
```diff
@@ -40,7 +42,9 @@
-#if !IS_ENABLED(CONFIG_SOC_NRF52810) && !IS_ENABLED(CONFIG_SOC_NRF52805)
+#if (!IS_ENABLED(CONFIG_SOC_NRF52810) &&	\
+	!IS_ENABLED(CONFIG_SOC_NRF52805) &&	\
+	!IS_ENABLED(CONFIG_SOC_NRF52811))
```
For nRF52811, we need to also add the following:
```
#if IS_ENABLED(CONFIG_SOC_NRF52811)
BUILD_ASSERT(!(SPI_ENABLED(1) && I2C_ENABLED(0)),
	     "Only one of the following peripherals can be enabled: "
	     "SPI1, SPIM1, SPIS1, TWI0, TWIM0, TWIS0. "
	     "Check nodes with status \"okay\" in zephyr.dts.");
#endif
```

mbolivar-nordic replied:
> For nRF52811, we need to also add the following:

Added, and tagged on a DTSI fix for this SoC I noticed while looking at it for this PR.

mbolivar-nordic committed:
dts: nrf52811: fix indendation in dtsi

Commit 233149eec5eb164abdeee4f3ff327fda976c37fc moved flash sram nodes
under /soc for various nordic ICs, but the indentation isn't right for
52811. Fix it.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic committed:
soc: nrf52811: add missing instance 3 check

spi1 and i2c0 share peripheral ID number 3.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>
